PYME/Acquire/ActionManager.py

```diff
@@ -121,5 +125,83 @@ def Tick(self, **kwargs):
                 self.isLastTaskDone = fcn(**args)
                 
 
+class ActionManagerWebWrapper(object):
+    def __init__(self, action_manager):
+        """ Wraps an action manager instance with server endpoints
+
+        Parameters
+        ----------
+        action_manager : ActionManager
+            action manager instance to wrap
+        """
+        self.action_manager = action_manager
+    
+    @webframework.register_endpoint('/queue_action', output_is_json=False)
+    def queue_action(self, body):
+        """
+        adds an action to the queue
+
+        Parameters
+        ----------
+        body: str
+            json.dumps(dict) with the following keys:
+                function_name : str
+                    The name of a function relative to the microscope object.
+                    e.g. to `call scope.spoolController.StartSpooling()`, you 
+                    would use a functionName of 'spoolController.StartSpooling'.
+                    
+                    The function should either return `None` if the operation 
+                    has already completed, or function which evaluates to True 
+                    once the operation has completed. See 
+                    `scope.spoolController.StartSpooling()` for an example.
+                args : dict, optional
+                    a dictionary of arguments to pass to `function_name`
+                nice : int, optional
+                    priority with which to execute the function, by default 10. 
+                    Functions with a lower nice value execute first.
+                timeout : float, optional
+                    A timeout in seconds from the current time at which the 
+                    action becomes irrelevant and should be ignored. By default
+                    1e6.
+        """
+        import json
+        params = json.loads(body)
+        function_name = params['function_name']
+        args = params.get('args', {})
+        nice = params.get('nice', 10.)
+        timeout = params.get('timeout', 1e6)
+        self.action_manager.QueueAction(function_name, args, nice, timeout)
+
+
+class ActionManagerServer(webframework.APIHTTPServer, ActionManagerWebWrapper):
```

Comment:
I don't like the idea of having multiple different servers running within PYMEAcquire, both due to the overhead, and the code duplication it entails. Medium term this will be addressed by having having `acquire_server` running the whole time (even with the wx frontend).

We can possibly have this here short term, but should be flagged that it will dissapear.

Reply:
OK, I'll flag it for now. Running the entire acquire_server the whole time felt like a big leap but maybe I overestimated it. Worth noting that combining the offsetpiezo stuff with the acquire server might also be nice re:reducing overhead.

Reply:
added the note.